Declare slash as a dependency of gatsby-plugin-offline

<!--
  Please fill out each section below, otherwise your issue will be closed. This info allows Gatsby maintainers to diagnose (and fix!) your issue as quickly as possible.

  Useful Links:
  - Documentation: https://www.gatsbyjs.org/docs/
  - How to File an Issue: https://www.gatsbyjs.org/contributing/how-to-file-an-issue/

  Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
-->

## Description

`slash` is being required by the `gatsby-node` script of `gatsby-plugin-offline`, however, it is not declared as a dependency in `package.json`.

### Steps to reproduce

* Use pnpm to install gatsby and gatsby-plugin-offline
* Include gatsby-plugin-offline as a plugin

### Expected result

No errors

### Actual result

`gatsby develop` fails to run with

`Error: Cannot find module 'slash'`

### Environment

Run `gatsby info --clipboard` in your project directory and paste the output here.

--- a/packages/gatsby-plugin-offline/package.json
+++ b/packages/gatsby-plugin-offline/package.json
@@ -11,6 +11,7 @@
     "cheerio": "^1.0.0-rc.2",
     "idb-keyval": "^3.1.0",
     "lodash": "^4.17.10",
+    "slash": "^3.0.0",
     "workbox-build": "^3.6.3"
   },
   "devDependencies": {

